app/controllers/state_file/questions/canceled_data_transfer_controller.rb

```diff
@@ -36,11 +36,8 @@ def redirect_if_no_us_state_in_params
         return if params[:us_state]
 
         if session[:state_file_intake].present?
-          if current_intake.class == StateFileAzIntake
-            redirect_link = az_questions_canceled_data_transfer_path(us_state: "az")
-          elsif current_intake.class == StateFileNyIntake
-            redirect_link = ny_questions_canceled_data_transfer_path(us_state: "ny")
-          end
+          state_code = StateFile::StateInformationService.state_code_from_intake_class(current_intake)
+          redirect_link = send("#{state_code}_questions_canceled_data_transfer_path".to_sym, us_state: state_code)
```

Comment:
I haven't tested this, but something like this should be possible and slightly more readable?

```suggestion
          redirect_link = StateFile::Questions::CanceledDataTransferController.to_path_helper(action: :cancel_data_transfer, us_state: state_code
```

Reply:
I started implementing your change, but realized this code is actually:
1. Not covered by any test
2. Unreachable and dead (The code tests what to do if there is no us_state param - but all routes require one:
`(/:locale)/:us_state/questions/canceled-data-transfer(.:format)`)

So I removed it.